sql/driver/wire.go

```diff
@@ -26,43 +29,34 @@ const (
 )
 
 func (d Datum) String() string {
-	if d.BoolVal != nil {
-		if *d.BoolVal {
-			return "true"
-		}
-		return "false"
-	}
-	if d.IntVal != nil {
-		return strconv.FormatInt(*d.IntVal, 10)
-	}
-	if d.FloatVal != nil {
-		return strconv.FormatFloat(*d.FloatVal, 'g', -1, 64)
-	}
-	if d.BytesVal != nil {
-		return string(d.BytesVal)
+	val := d.GetValue()
```

Comment:
Have you looked at the implementation of `GetValue()`? It just does the if-statements we were previously doing, but not you're doing those as well as a type switch. Perhaps something like the following:

```
  switch  {
    case d.BoolVal != nil:
      ...
    case d.IntVal != nil:
      ...
    case d.FloatVal != nil:
      ...
  }
```